### PR TITLE
migrate from dep to Go modules

### PR DESCRIPTION
**Pull request by fiunchinho**
I've used version `v0.16.7` for kubernetes components. I'm not sure which one we should use.

### CODE REVIEW

**Review by rossf7 — COMMENTED** [review]
Nice!

But on the k8s version in @giantswarm/wg-go-modules we decided to go with 1.16.6. e.g. https://github.com/giantswarm/chart-operator/pull/351

**Comment by fiunchinho** [comment]
Ok, I just changed the version. I was trying to create a new operator and I thought about migrating. I didn't want to step on anyone's toe!

File: go.mod [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+	golang.org/x/net v0.0.0-20191209160850-c0dbc17a3553 // indirect
+	golang.org/x/oauth2 v0.0.0-20191202225959-858c2ad4c8b6 // indirect
+	golang.org/x/sys v0.0.0-20191210023423-ac6580df4449 // indirect
+	golang.org/x/time v0.0.0-20191024005414-555d28b269f0 // indirect
+	golang.org/x/xerrors v0.0.0-20191204190536-9bdfabe68543 // indirect
+	google.golang.org/appengine v1.6.5 // indirect
+	k8s.io/api v0.16.6
+	k8s.io/apimachinery v0.16.6
+	k8s.io/client-go v0.16.6
+	k8s.io/kubernetes v1.17.3 // indirect
```
Comment (rossf7) [inline go.mod]: Shouldn't this also be for 1.16.6?

**Comment by rossf7** [comment]
> I was trying to create a new operator and I thought about migrating. I didn't want to step on anyone's toe!

@fiunchinho I think its good to migrate as new operators should use modules. But there are some docs / notes in the #wg-go-modules channel that might help.

**Comment by fiunchinho** [comment]
Migrated microerror.Stack() calls to microerror.JSON()

**Review by MarcelMue — APPROVED** [review]
Fine as a first step for me.

**Comment by fiunchinho** [comment]
> Fine as a first step for me.

is there anything left to do?

**Comment by MarcelMue** [comment]
> > Fine as a first step for me.
> 
> is there anything left to do?

There is some work in wg-go-modules where we need to update projects and this will then change here. Nothing to do here right now IMO.

File: vendor/github.com/coreos/go-semver/NOTICE [hunk 2]
```diff
@@ -0,0 +1,5 @@
+CoreOS Project
```
Comment (kopiczko) [inline vendor/github.com/coreos/go-semver/NOTICE]: Why do we still have vendor in here?

Comment (fiunchinho) [reply vendor/github.com/coreos/go-semver/NOTICE]: Good point. I guess we can delete it now.

**Pull request merged**